### DISCUSSION TITLE
How to make use of rhvoice.conf?

### DISCUSSION THREAD

**Discussion by Kzer-Za** [RHVoice/RHVoice#796] [Q&A]
Okay, at first I misunderstood and thought that the conf file should affect RHVoice-test (https://github.com/RHVoice/RHVoice/issues/795). But now I have configured speech-dispatcher, but spd-say still uses the wrong voice.

My RHVoice.conf:

```
quality=max

default_volume=1.5

voices.vitaliy.default_pitch=1.0
voices.bdl.default_pitch=1.2

voices.vitaliy.default_rate=0.85
voices.bdl.default_rate=0.8

voice_profiles=vitaliy+bdl
```

When I run `spd-say -L` it does list all the profiles available for RHVoice (imcluding Vitaliy+Bdl, as specified in the conf file):

```
                     NAME                 LANGUAGE                  VARIANT
                     Alan                       en                     none
             Aleksandr-hq                       ru                     none
                     Anna                       ru                     none
                  Artemiy                       ru                     none
                      Bdl                       en                     none
                      Clb                       en                     none
              Evgeniy-Eng                       en                     none
              Evgeniy-Rus                       ru                     none
                   Lyubov                       en                     none
                  Mikhail                       ru                     none
                    Pavel                       ru                     none
                      Slt                       en                     none
                  Vitaliy                       ru                     none
              Vitaliy+Bdl                       ru                     none
                    Yuriy                       ru                     none
```

But if I run `spd-say "some text"`, it still says the text in the wrong voice. E.g. if I run `spd-say "Проверка русской речи"`, the voice used is Aleksandr-hq, with default pitch, rate, and volume.

My speechd.conf: https://dpaste.org/SjhMa

**Comment by zstanecic** [comment]
Voice names are case sensitive. vitaliy Is Vitaliy. 

 

 

From: Denis ***@***.***> 
Sent: Monday, October 2, 2023 12:05 PM
To: RHVoice/RHVoice ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [RHVoice/RHVoice] How to make use of rhvoice.conf? (Discussion #796)

 

Okay, at first I misunderstood and thought that the conf file should affect RHVoice-test (#795 <https://github.com/RHVoice/RHVoice/issues/795> ). But now I have configured speech-dispatcher, but spd-say still uses the wrong voice.

My RHVoice.conf:

quality=max
 
default_volume=1.5
 
voices.vitaliy.default_pitch=1.0
voices.bdl.default_pitch=1.2
 
voices.vitaliy.default_rate=0.85
voices.bdl.default_rate=0.8
 
voice_profiles=vitaliy+bdl

When I run spd-say -L it does list all the profiles available for RHVoice (imcluding Vitaliy+Bdl, as specified in the conf file):

                     NAME                 LANGUAGE                  VARIANT
                     Alan                       en                     none
             Aleksandr-hq                       ru                     none
                     Anna                       ru                     none
                  Artemiy                       ru                     none
                      Bdl                       en                     none
                      Clb                       en                     none
              Evgeniy-Eng                       en                     none
              Evgeniy-Rus                       ru                     none
                   Lyubov                       en                     none
                  Mikhail                       ru                     none
                    Pavel                       ru                     none
                      Slt                       en                     none
                  Vitaliy                       ru                     none
              Vitaliy+Bdl                       ru                     none
                    Yuriy                       ru                     none

But if I run spd-say "some text", it still says the text in the wrong voice. E.g. if I run spd-say "Проверка русской речи", the voice used is Aleksandr-hq, with default pitch, rate, and volume.

My speechd.conf: https://dpaste.org/SjhMa

**Reply by Kzer-Za** [reply]
I have modified the conf file so that voices start with the capital letters, but it didn't help. `spd-say "some text"` still uses Aleksandr-hq for Russian and Alan for English.

**Comment by zstanecic** [comment]
You need to switch voice.

**Reply by Kzer-Za** [reply]
You mean use the `spd-say -y somevoice`? But then why have a config file at all? Also, even if I use the needed voices via this switch, their pitch and rate are still default (1.0), and not those specified in the config file.

**Comment by zstanecic** [comment]
You need to switch the voice, because you are setting up voice profiles in the configuration file.

You are specifying these for the use by the speech dispatcher or any other client that walks and talks on earth with RHVoice.

Default rate is the rate specified by the incremental values of 50 percent. So, rate should be specified, too.

Specifying deult rate means specifying rate of 50 percent

**Reply by Kzer-Za** [reply]
So, what, RHVoice.conf is unused? Since you have to specify everything with command-line switches anyway?

**Comment by zstanecic** [comment]
No, it is used, but in different way you expect.

**Comment by Kzer-Za** [comment]
Okay, maybe it's better to phrase my question from the desired result perspective.

I have a text mostly in Russian but with insterspersed English phrases. Splitting it into separate Russian and English parts would be ineffective because English sections are very short --- like I said, only phrases and even separate words. So I need to process the whole text with 2 voices: a Russian voice and an English voice. However, I would also prefer to adjust the pitch and rate for these voices separately: for English voice (Bdl) 1.2 and 0.8, and for Russian voice (Vitaliy) 1.0 and 0.85.

Doing so by command-line switches for RHVoice-test is impossible because it applies the same pitch and rate specified by -t and -r switches to both voices; and like I said, I need different settings per voice.

I thought I could use RHVoice.conf for this purpose because in there you can specify settings per voice, but that isn't working.

So, is it possible to transfer text to audio, using different voices for Russian and English and appying different pitch and rate for these voices?
